javascript/javascript2/week2/movies.js

```diff
@@ -0,0 +1,74 @@
+const { movies } = require("./movies-data.js");
+//Create an array of movies containing the movies with a short title
+const shortTitlemovies = movies.filter((movie) => movie.title.length <= 10);
+console.log(shortTitlemovies.length);
+
+//Create an array of movie titles with long movie titles
+const longMovieTitles = movies
+  .filter((movie) => movie.title.length > 10)
+  .map((movie) => movie.title);
+console.log(longMovieTitles);
+
+//Count the number of movies made between 1980-1989 (including both the years)
+const numberOfMovies = movies.filter(
+  (movie) => movie.year >= 1980 && movie.year <= 1989
+).length;
+console.log(" Movies from 1980 to 1989 : " + numberOfMovies);
+
+//Create a new array that has an extra key called tag. The tag is based on the rating: Good (>= 7), Average (>= 4 and < 7), Bad (< 4)
+const moviesWithTag = movies.filter((movie) => {
+  if (movie.rating >= 7) movie.tag = "Good";
+  if (movie.rating >= 4 && movie.rating < 7) movie.tag = "Average";
+  if (movie.rating < 4) movie.tag = "Bad";
+  return movie;
+});
```

Comment:
This is a bit tricky, but you are actually changing the original array! You are also filtering but not returning a boolean

Reply:
hope this will work.
function reducer(accumulator, movie) {
  const currentmovie = {};
  Object.assign(currentmovie, movie);
  if (currentmovie.rating >= 7) currentmovie.tag = "Good";
  if (currentmovie.rating >= 4 && currentmovie.rating < 7)
    currentmovie.tag = "Average";
  if (currentmovie.rating < 4) currentmovie.tag = "Bad";
  accumulator.push(currentmovie);
  return accumulator;
}
const moviesWithTag = movies.reduce(reducer, []);
console.log(moviesWithTag);

Reply:
Yes you can definitely do it that way!

Reply:
thanks